pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ClientCnx.java

```diff
@@ -61,6 +64,13 @@
 
     private final CompletableFuture<Void> connectionFuture = new CompletableFuture<Void>();
     private final Semaphore pendingLookupRequestSemaphore;
+    private final Timer timer;
```

Comment:
that's true, then we could use the `eventLoop` associated with the netty channel to schedule the task.
(If we use a timer per connection, it will create a different thread for each of them)